src/runtime/HalideRuntime.h

```diff
@@ -765,4 +800,82 @@ extern double halide_float16_bits_to_double(uint16_t);
 } // End extern "C"
 #endif
 
+#ifdef __cplusplus
```

Comment:
I'm all for encapsulating at the Type level. I was likely erring on the
side of not changing the compiler code too much because it seemed
indicative of a bad change, but given this input I'll pursue that approach.

I think the constructors were necessary to have Type inherit from
halide_type_t.
If we do the encapsulation, those can be nuked. Bit fields are in the same
boat as I did that because a test failed do to a change in behavior with no
compile error and that bothered me. (C++'s type system conflating
characterness with bit size and then having overloading is really not a
good thing.)

On Fri, Nov 13, 2015 at 2:34 PM, Andrew Adams <notifications@github.com>
wrote:

> In src/runtime/HalideRuntime.h
> <https://github.com/halide/Halide/pull/973#discussion_r44842371>:
>
> > @@ -765,4 +800,82 @@ extern double halide_float16_bits_to_double(uint16_t);
> >  } // End extern "C"
> >  #endif
> >
> > +#ifdef __cplusplus
>
> This C++ block looks like useful syntactic sugar to me. I think it's worth
> having in HalideRuntime.h
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/halide/Halide/pull/973/files#r44842371>.
>